api/src/services/users.js

```diff
@@ -623,9 +868,82 @@ module.exports = {
               rev: resp.rev
             };
           })
+          .then(() => manageTokenLogin(data, response))
           .then(() => response);
       });
   },
 
-  DOC_IDS_WARN_LIMIT
+  DOC_IDS_WARN_LIMIT,
+
+  /**
+   * Searches for a user with active, not expired token-login that matches the requested token and hash.
+   *
+   * @param {String} token
+   * @param {String} hash
+   * @returns {Promise<string>} the id of the matched user
+   */
+  getUserByToken: (token, hash) => {
+    if (!token || !hash) {
+      return Promise.resolve(false);
+    }
+
+    return db.users.query('token-login/users-by-token', { key: [token, hash] }).then(response => {
```

Comment:
I feel uncomfortable adding views to the _users db... but I don't know why. Good idea storing the token in the _users db for security purposes.

Instead of adding a view can we find the _users UUID with the hash and use that to get the doc and then look the token and expiry up?